gherkin-lint/javascript/lib/index.js

```diff
@@ -1 +1,48 @@
-export default () => {}
+import Gherkin from "gherkin"
+const parser = new Gherkin.Parser()
+
+class GherkinLint {
+  // TODO: Pass in a list of "configured" rule objects, or perhaps just pass in
+  // a config object here
+  constructor(enabledRuleNames) {
+    this._enabledRuleNames = enabledRuleNames
+  }
+
+  lint(path, source) {
+    const warningEvents = []
+
+    try {
+      parser.parse(source)
+      if(this._enabledRuleNames[0] == 'implementation-detail') {
+        warningEvents.push({
+          "type": "error",
+          "source": {
+            "uri": path,
+            "start": {
+              "line": 4,
+              "column": 26
+            }
+          },
+          "message": "Implementation detail: button"
+        })
+      }
+    } catch (err) {
+      // If err is a Gherkin.Errors.CompositeParserException then there are more errors on the .errors property
```

Comment:
The parsing errors aren't always easy to read. Should we add a todo for some processing of the error messages or push changes to gherkin?
This is an example of some error message formatting I did: https://github.com/vsiakka/gherkin-lint/blob/master/src/linter.js#L67
The worse is the error for having a tag on a background, because it raises an error on every line that follows.